### ISSUE TITLE
no-global-function-names doesn't report errors inside of `@if` statements.. it works fine otherwise

### ISSUE THREAD

**Issue by helaoutar** [stylelint/stylelint#5929]
### What steps are needed to reproduce the bug?

Run Stylint on the following `scss` file:

```scss
@mixin is($breakpoint) {
  // No error reported!
  @if map-has-key($breakpoints, $breakpoint) { 

   // Getting an error as expected
    color: map-get($colors, red);   

    @include _media($from: $breakpoint, $until: map.get($next, $breakpoint)) {
      @content;
    }
  }
}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": ["stylelint-config-standard-scss", "stylelint-config-prettier-scss"],
  "rules": {
    "string-quotes": "single",
    "function-name-case": null,
    "scss/at-mixin-pattern": null,
    "scss/dollar-variable-pattern": null,
    "scss/at-function-pattern": null
  }
}

```

### How did you run Stylelint?

CLI with `"stylelint \"**/*.scss\"`

### Which version of Stylelint are you using?

`14.5.0`

### What did you expect to happen?

I was expecting an error, since I'm using `map-has-keys` instead of `map.has-keys`

### What actually happened?

No error is reported
